### PR TITLE
CR for Job #20370 - Talk bubble fix - from a suggestion in https://worklist.net/20361

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Code Review for Job #20370 - Workitem available at https://www.worklist.net/20370

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 21f12c1d6c9d458c9c98376f323ff1581ac32f3b available at: https://gist.github.com/abf099db384e76a23093<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4369.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4369.dmg

File: libraries/render-utils/src/GeometryCache.cpp [hunk 1]
```diff
@@ -53,18 +53,33 @@ const int NUM_COORDS_PER_VERTEX = 3;
...
     IntPair slicesStacksKey(slices, stacks);
-    Vec3Pair colorKey(glm::vec3(color.x, color.y, slices), glm::vec3(color.z, color.y, stacks));
+    Vec3Pair colorKey(glm::vec3(color.x, color.y, slices), glm::vec3(color.z, color.w, stacks));
 
     int vertices = slices * (stacks - 1) + 2;    
     int indices = slices * (stacks - 1) * NUM_VERTICES_PER_TRIANGULATED_QUAD;
     
-    if (!_sphereVertices.contains(radiusKey)) {
+    if (registered && (!_registeredSphereVertices.contains(id) || _lastRegisteredSphereVertices[id] != radiusKey)
+        || !registered && !_sphereVertices.contains(radiusKey)) {
```
Comment (Atlante45) [inline libraries/render-utils/src/GeometryCache.cpp]: @ctrlaltdavid I believe you are missing some parenthesis here.
Xcode prints a warning, I assume you want  .. || (!registered && !_sphereVertices.contains(radiusKey)) ...

File: libraries/render-utils/src/GeometryCache.cpp [hunk 2]
```diff
@@ -106,10 +121,29 @@ void GeometryCache::renderSphere(float radius, int slices, int stacks, const glm
...
     }
+    #ifdef WANT_DEBUG
+    else if (registered) {
+        qDebug() << "renderSphere()... REUSING PREVIOUSLY REGISTERED VERTICES BUFFER";
+    }
+    #endif
     
-    if (!_sphereIndices.contains(slicesStacksKey)) {
+    if (registered && (!_registeredSphereIndices.contains(id) || _lastRegisteredSphereIndices[id] != slicesStacksKey)
+        || !registered && !_sphereIndices.contains(slicesStacksKey)) {
```
Comment (Atlante45) [inline libraries/render-utils/src/GeometryCache.cpp]: same here

File: libraries/render-utils/src/GeometryCache.cpp [hunk 3]
```diff
@@ -173,10 +207,29 @@ void GeometryCache::renderSphere(float radius, int slices, int stacks, const glm
...
         #endif
     }
+    #ifdef WANT_DEBUG
+    else if (registered) {
+        qDebug() << "renderSphere()... REUSING PREVIOUSLY REGISTERED INDICES BUFFER";
+    }
+    #endif
+
+    if (registered && (!_registeredSphereColors.contains(id) || _lastRegisteredSphereColors[id] != colorKey) 
+        || !registered && !_sphereColors.contains(colorKey)) {
```
Comment (Atlante45) [inline libraries/render-utils/src/GeometryCache.cpp]: same here

File: libraries/render-utils/src/GeometryCache.cpp (hunk 1 above)
Comment (Atlante45) [inline libraries/render-utils/src/GeometryCache.cpp]: Or maybe you wanted this, make more sense.
(registered && (!_registeredSphereVertices.contains(id) || _lastRegisteredSphereVertices[id] != radiusKey)) || (!registered && !_sphereVertices.contains(radiusKey))

**Comment by ctrlaltdavid** [comment]
Thanks @Atlante45 ... Addressed in a new PR.

**Pull request merged**